test/unit/lib/sources.js

```diff
@@ -0,0 +1,190 @@
+/* eslint-env mocha */
+"use strict";
+
+const assert = require('proclaim');
+const mockery = require('mockery');
+const path = require('path');
+const sinon = require('sinon');
+require('sinon-as-promised');
+
+describe('lib/sources', () => {
+	let aliases;
+	let denodeify;
+	let fs;
+	let sources;
+	let process;
+	let consoleMock;
+
+	beforeEach(() => {
+		denodeify = require('../mock/denodeify.mock');
+		mockery.registerMock('denodeify', denodeify);
+
+		fs = require('../mock/fs.mock');
+		mockery.registerMock('fs', fs);
+		fs.readdirSync.returns([]);
+
+		process = require('../mock/process.mock');
+		mockery.registerMock('process', process);
+
+		consoleMock = require('../mock/console.mock');
+		mockery.registerMock('console', consoleMock);
+
+		aliases = {};
+		mockery.registerMock('../polyfills/__dist/aliases.json', aliases);
+	});
+
+	it('exports an object', () => {
+		sources = require('../../../lib/sources');
+		assert.isObject(sources);
+	});
+
+	it('has a polyfillExistsSync method', () => {
+		sources = require('../../../lib/sources');
+		assert.isFunction(sources.polyfillExistsSync);
+	});
+
+	it('has a getPolyfillMetaSync method', () => {
+		sources = require('../../../lib/sources');
+		assert.isFunction(sources.getPolyfillMetaSync);
+	});
+
+	it('has a listPolyfillsSync method', () => {
+		sources = require('../../../lib/sources');
+		assert.isFunction(sources.listPolyfillsSync);
+	});
+
+	it('has a listPolyfills method', () => {
+		sources = require('../../../lib/sources');
+		assert.isFunction(sources.listPolyfills);
+	});
+
+	it('has a getConfigAliasesSync method', () => {
+		sources = require('../../../lib/sources');
+		assert.isFunction(sources.getConfigAliasesSync);
+	});
+
+	it('has a streamPolyfillSource method', () => {
+		sources = require('../../../lib/sources');
+		assert.isFunction(sources.streamPolyfillSource);
+	});
+
+	it('has a getPolyfill method', () => {
+		sources = require('../../../lib/sources');
+		assert.isFunction(sources.getPolyfill);
+	});
+
+	it('loads the polyfill aliases file', () => {
+		// TODO
+	});
+
+	it('reads the polyfill directory', () => {
+		sources = require('../../../lib/sources');
+		assert.calledWithExactly(fs.readdirSync, path.join(__dirname, '../../../polyfills/__dist'));
+	});
+
+	it('filters out json files from the polyfill directory', () => {
+		const spy = sinon.spy(Array.prototype, 'filter');
+		sources = require('../../../lib/sources');
+		spy.restore();
+		assert.equal(spy.lastCall.args[0]('aliases.json'), false);
+		assert.equal(spy.lastCall.args[0]('example.json'), false);
+	});
+
+	it('catches errors when reading the polyfill directory, logs a friendly error message and exits', () => {
+		const error = new Error;
+		fs.readdirSync.throws(error);
+		assert.doesNotThrow(() => require('../../../lib/sources'));
+		assert.calledWithExactly(consoleMock.log, "No polyfill sources found.  Run `npm run build` to build them");
```

Comment:
Asserting the exact error text seems a tad pedantic - maybe enough to confirm that we log something?